File: python/달팽이(1913).py
N = int(input())
find_num = int(input())

# 하, 우, 상, 좌
dx = [1, 0, -1, 0]
dy = [0, 1, 0, -1]

graph = [[0]*(N) for _ in range(N)]

# 시작 위치 설정
x, y = 0, 0
num = N**2
graph[x][y] = num

# 찾을 값 위치
find_x, find_y = 0, 0

# 초기 방향
dir_idx = 0

while num > 1:
    nx = x + dx[dir_idx]
    ny = y + dy[dir_idx]

    if 0 <= nx < N and 0 <= ny < N and graph[nx][ny] == 0:
        num -= 1
        graph[nx][ny] = num
        x, y = nx, ny # x, y값 업데이트

        if num == find_num:
            find_x = nx
            find_y = ny

    else:
        dir_idx = (dir_idx + 1) % 4

for row in graph:
    print(*row)

print(find_x+1, find_y+1)

File: js/달팽이(1913).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const N = Number(input[0]);
const findNum = Number(input[1]);

const graph =  Array.from({length: N}, () => Array(N).fill(0));

// 방향 : 하, 우, 상, 좌
const dx = [1, 0, -1, 0]; 
const dy = [0, 1, 0, -1];

// 첫 위치
let x = 0;
let y = 0;
let num = N**2;
graph[x][y] = num;

let findX = 0;
let findY = 0;

let dir_idx = 0;

while (num > 1) {
    const nx = x + dx[dir_idx];
    const ny = y + dy[dir_idx];

    if (0 <= nx && nx < N && 0 <= ny && ny < N && graph[nx][ny] === 0) {
        num -= 1;
        graph[nx][ny] = num;
        x = nx;
        y = ny;

        if (num === findNum) {
            findX = nx;
            findY = ny;
        }
    } else {
        dir_idx = (dir_idx + 1) % 4;
    }
}

graph.forEach(row => console.log(row.join(' ')));
console.log(findX, findY);